api/src/main/java/uk/gov/dvsa/mot/persist/jdbc/MySqlConnectionFactory.java

```diff
@@ -17,24 +20,57 @@
 public class MySqlConnectionFactory implements ConnectionFactory {
     private static final Logger logger = Logger.getLogger(MySqlConnectionFactory.class);
 
+    private static final DatabasePasswordHolder DB_PASSWORD_HOLDER = new DatabasePasswordHolder();
+
     @Override
     public Connection getConnection() {
 
         try {
-            String url = ConfigManager.getEnvironmentVariable(ConfigKeys.DatabaseConnection);
-            String username = ConfigManager.getEnvironmentVariable(ConfigKeys.DatabaseUsername);
-
-            String password = ConfigManager.getEnvironmentVariable(ConfigKeys.DatabaseEncryptedPassword);
-
-            if (password == null) {
-                password = ConfigManager.getEnvironmentVariable(ConfigKeys.DatabasePassword);
-            }
-
-            return DriverManager.getConnection(url, username, password);
+            return DriverManager.getConnection(getDbUrl(), getDbUsername(), getDbPassword());
         } catch (Exception e) {
             logger.error("Unable to connect to database", e);
             throw new InternalException(e);
         }
     }
 
+    private String getDbUrl() throws IOException {
+
+        return ConfigManager.getEnvironmentVariable(ConfigKeys.DatabaseConnection);
+    }
+
+    private String getDbUsername() throws IOException {
+
+        return ConfigManager.getEnvironmentVariable(ConfigKeys.DatabaseUsername);
+    }
+
+    private String getDbPassword() throws IOException {
+
+        return DB_PASSWORD_HOLDER.getDbPassword();
+    }
+
+    private static class DatabasePasswordHolder {
```

Comment:
Would it be better to have this split out into a separate class?  Would make unit testing of this class much easier :)